x-pack/legacy/plugins/siem/public/components/ml_popover/jobs_table/job_switch.tsx

```diff
@@ -48,17 +48,18 @@ export const JobSwitchComponent = ({
 }: JobSwitchProps) => {
   const [isLoading, setIsLoading] = useState(false);
   const handleChange = useCallback(
-    e => {
+    async e => {
       setIsLoading(true);
-      onJobStateChange(job, job.latestTimestampMs || 0, e.target.checked);
+      await onJobStateChange(job, job.latestTimestampMs || 0, e.target.checked);
```

Comment:
This fixes a bug where, after toggling an ML job, that EuiSwitch component is stuck in a loading state until the page is reloaded or the component is reinitialized. There's probably a more robust solution here with `useReducer` but I'm not sure that we need it.

Reply:
🦅 👀's @rylnd! Nice catch -- thank you! 🙂 

This gets traced back up to `ml_popover` where `enableDatafeed` is declared as `async`, but the cascading types are _not_.

https://github.com/elastic/kibana/blob/6ce17885eb1020a00749723ddd293dbd71eecba4/x-pack/legacy/plugins/siem/public/components/ml_popover/ml_popover.tsx#L105 

If we update the types in `job_switch` and `jobs_table` to be:

``` ts
onJobStateChange: (job: SiemJob, latestTimestampMs: number, enable: boolean) => Promise<void>;
```

we'll then get the appropriate linter error.

Reply:
Fixed in 1118794